packages/blockchain-tools/src/interfaces/Web3Server.ts

```diff
@@ -391,7 +425,7 @@ export default class Web3Server extends InterfaceBaseClass {
       body,
     }
     try {
-      const res = await fetch(this.network.rpcUrls.main, opt)
+      const res = await fetch(this?.network?.rpcUrls?.main || "", opt)
```

Comment:
This.network isn't optional, and `this.network.rpcUrls.main` should always be defined